### PR TITLE
Update Github workflows and Go dependencies

### PR DESCRIPTION
**Pull request by embano1**
This PR updates the Github workflows and Go dependencies to more recent versions:

- [x] Bump `go` directive in respective modules to `go 1.22`
- [x] Github workflows run with latest Go version (fixed to current 1.23)
- [x] Update linter and fix errors
- [x] Update README to indicate tested Go version for this SDK

Note that I didn't feel the need to bump modules straight to `go 1.23` because we don't use the latest language features and `go 1.22` seemed sufficient/causing less compatibility issues (or warnings) for users < Go 1.22.

### CODE REVIEW

**Comment by embano1** [comment]
@duglin for review

File: v2/protocol/http/write_request.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (b *httpRequestWriter) setBody(body io.Reader) error {
 		default:
 			// This is where we'd set it to -1 (at least
 			// if body != NoBody) to mean unknown, but
-			// that broke people during the Go 1.8 testing
+			// that broke people during the Go 1.8 testin2
```
Comment (duglin) [inline v2/protocol/http/write_request.go:89]: Typo?

Comment (embano1) [reply v2/protocol/http/write_request.go:89]: yeah, but not by me :)

I need to file a README update, lemme quickly address

Comment (embano1) [reply v2/protocol/http/write_request.go]: @duglin fixed

**Review by duglin — APPROVED** [review]

**Pull request merged**